src/test/kotlin/no/nb/mlt/wls/product/controller/ProductControllerTest.kt

```diff
@@ -65,7 +69,7 @@ class ProductControllerTest(
     @WithMockUser
     fun `createProduct with valid payload creates product`() {
         every {
-            synqProductService.createProduct(any())
+            runBlocking { synqProductService.createProduct(any()) }
```

Comment:
Have you just tried calling block() instead? Also if we need more complex testing then I think it would be good to use either a step verifier especially when/if we work with fluxes:
https://www.baeldung.com/reactive-streams-step-verifier-test-publisher
Kotlin seems to have a nice extension for it:
https://stackoverflow.com/a/66213441

Reply:
Originally when writing this I was running into issues where making blocking calls wasn't allowed. Not sure exactly what the context was, but it seems like this is now possible. Will do that

Reply:
Double-checked why this was implemented this way and found the reasoning:
You aren't meant to block objects that return Reactive elements, and as such if you are on a thread you can get an error message like this: 
`java.lang.IllegalStateException: block()/blockFirst()/blockLast() are blocking, which is not supported in thread parallel-2`
This happens if you try blocking any `Mono` from a `ReactiveMongoRepository`, which makes sense. Using this makes Mongo send out their objects in a reactive manner, so blocking them will just make it act as a regular `MongoRepository` - entirely defeating the purpose to begin with. 
Because I ran into this issue first, I didn't consider blocking the SynQ services as I thought I would run into the same issue. This assumption is wrong, as SynQ by nature is a blocking application, and therefore we can block it.